Win32-network/cbits/Win32Async.c

```diff
@@ -140,10 +140,16 @@ void HsAsyncWrite(HANDLE handle, void *buf, int size,
     }
 }
 
-void HsAssociate(HANDLE handle, HANDLE port)
+BOOL HsAssociateHandle(HANDLE handle, HANDLE port)
 {
-  CreateIoCompletionPort(handle, port, MAGIC_COMPLETION_KEY, 0);
-  return;
+  return (CreateIoCompletionPort(handle, port, MAGIC_COMPLETION_KEY, 0)
+          == port);
+}
+
+BOOL HsAssociateSocket(SOCKET socket, HANDLE port)
+{
+  return (CreateIoCompletionPort((HANDLE) socket, port, MAGIC_COMPLETION_KEY, 0)
```

Comment:
Is there any MSDN doc we can point to to explain that a socket is really just a handle and so casting is legitimate?

Reply:
Yes, there is a comment in MSD docs which states that `CreateIOCompletionPort` can accept a socket, I'll dig it out.

Reply:
This is documented at the top of `CreateIoCompletionPort`
https://docs.microsoft.com/en-us/windows/win32/fileio/createiocompletionport#createiocompletionport-function
I added this reference to the comment.